sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -806,7 +834,7 @@ uint8_t gcoap_op_state(void)
 int gcoap_get_resource_list(void *buf, size_t maxlen, uint8_t cf)
 {
     assert(cf == COAP_CT_LINK_FORMAT);
-#ifndef DEVELHELP
+#ifdef NDEBUG
```

Comment:
why this change?

Reply:
The definition of `assert` depends on the definition of `NDEBUG`
`NDEBUG` is defined by the build system if `DEVELHELP`is defined with `-DDEVELHELP`.
https://github.com/RIOT-OS/RIOT/blob/2017.07/makefiles/cflags.inc.mk#L55

So this `ifdef` worked because of the build system, and if you do `CFLAGS += -DDEVELHELP=1`, `assert` is not defined and you get an unused variable warning.

It may belong in another PR though.